src/components/Gnb.jsx

```diff
@@ -0,0 +1,59 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import styles from "./Gnb.module.css";
+import logo from "/src/assets/logo.png";
+import { getUserData } from "../api";
+import useAsync from "../hooks/useAsync";
+
+function LoginButton() {
+  return (
+    <Link to="/signin" className={styles.loginBtn}>
+      로그인
+    </Link>
+  );
+}
+
+function UserProfile() {
+  const [user, setUser] = useState(null);
+  const [isApplying, ApplyingError, onApplyAsync] = useAsync(getUserData);
+
+  const applyUserData = useCallback(async () => {
+    const userData = await onApplyAsync();
+    if (!userData) return;
+    setUser(userData);
+  }, [onApplyAsync]);
+
+  useEffect(() => {
+    applyUserData();
+  }, [applyUserData]);
+
+  if (ApplyingError) return <div>{ApplyingError.message}</div>;
+  return (
+    user && (
+      <div className={styles.userContainer}>
+        <img
+          className={styles.userImg}
+          src={user.profileImageSource}
+          alt={user.name}
+        />
+        <p className={styles.userEmail}>{user.email}</p>
+      </div>
+    )
+  );
+}
+
+function Gnb() {
```

Comment:
```suggestion
function GNB() {
```
(참고만) GlobalNavigationBar이니 GNB는 어떨까요?ㅋㅋ

Reply:
이런 디테일 너무 감사합니다ㅎㅎㅎ